cluster/gce/gci/configure-helper.sh

```diff
@@ -1937,6 +1947,15 @@ function start-kube-controller-manager {
   sed -i -e "s@{{flexvolume_hostpath}}@${FLEXVOLUME_HOSTPATH_VOLUME}@g" "${src_file}"
   sed -i -e "s@{{cpurequest}}@${KUBE_CONTROLLER_MANAGER_CPU_REQUEST}@g" "${src_file}"
 
+  if which setfacl &> /dev/null ; then
```

Comment:
I am concerned that this will fail silently when `setfacl` is not found on the image, but we requested that `controller-manager` runs as non-root. I think we should consider such case as a failed pre-condition and fail hard.
I suggest that we check if `KUBE_CONTROLLER_MANAGER_RUNASUSER` is set (this implies that we explicitly requested this functionality), and fail hard if `setfacl` is not found on the image.

Reply:
I didn't want to be too strict here as there are several factors that need to align for us to run kube-controller-manager as non root. 

- `KUBE_CONTROLLER_MANAGER_RUNASUSER` needs to be present. (this is yet to be checked in) 
- COS image must have `setfacl` (this is yet to be checked in) 
- this CL must be checked in and possibly back-ported.

 With all this it is hard to predict what the versions and the timelines will be. So I went with an approach that is forgiving and does not break cluster creation when one of the requirements is not present.  

Checking for `KUBE_CONTROLLER_MANAGER_RUNASUSER` is not strictly needed as we can default to 2017 when it's not set. So the only requirement is `setfacl` being installed.

Is your concern that the check `if which setfacl &> /dev/null ; then` not sufficient? I tested it with images with and without `setfacl` and it worked as expected in my gke sandbox.

Reply:
Is it possible to check the version of the COS image (and that we are running on COS)?
Concretely, run controller-manager as non-root only when COS image is above certain version?

Reply:
To do this reliably we would have to do this in cluster_server where we check the master image version and if it's greater that what we want we could add `KUBE_CONTROLLER_MANAGER_RUNASUSER` to `kube_env`? What do you think about that approach?

Reply:
sgtm

Out of curiosity, is is possible to deduce the image version from this:
```

cat /etc/os-release
BUILD_ID=12371.183.0
NAME="Container-Optimized OS"
KERNEL_COMMIT_ID=0c94af8a1221168be9c750ad378d494e969fa425
GOOGLE_CRASH_ID=Lakitu
VERSION_ID=77
BUG_REPORT_URL="https://cloud.google.com/container-optimized-os/docs/resources/support-policy#contact_us"
PRETTY_NAME="Container-Optimized OS from Google"
VERSION=77
GOOGLE_METRICS_PRODUCT_ID=26
HOME_URL="https://cloud.google.com/container-optimized-os/docs"
ID=cos
```

Reply:
Yes it is possible to deduce the version from the file.